assets/scripts/app/DebugInfo.jsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * DebugInfo.jsx
+ *
+ * Displays a debugging overlay that shows the current state of the application.
+ *
+ * @module DebugInfo
+ * @requires keypress
+ */
+import React from 'react'
+import { cloneDeep } from 'lodash'
+import { getStreet } from '../streets/data_model'
+import { getUndoStack } from '../streets/undo_stack'
+import { getSettings } from '../users/settings'
+import { registerKeypress, deregisterKeypress } from './keypress'
+import { loseAnyFocus } from './focus'
+
+export default class DebugInfo extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      visible: false,
+      content: ''
+    }
+
+    this.showDebugInfo = this.showDebugInfo.bind(this)
+    this.hideDebugInfo = this.hideDebugInfo.bind(this)
+  }
+
+  componentDidMount () {
+    // Register keyboard input for show (shift-D)
+    registerKeypress('shift d', this.showDebugInfo)
```

Comment:
This might be thinking too far ahead, but what do you think about unregistering listeners in componentWillUnmount() ? Obviously, how the app is coded right now, this won't get detached from the DOM but it could in the future??